src/term/mod.rs

```diff
@@ -202,6 +202,9 @@ pub struct Term {
     /// The grid
     grid: Grid<Cell>,
 
+    /// Wrap on next input
+    wrap: bool, 
```

Comment:
There's a lot going on with this flag. Can we document how it's used here? Maybe just pull in some of what you wrote in the PR. Or maybe it should be documented inline with `input()`? Anyway, it would be nice to have a summary to help understand this feature without reading the code.